lib/modules/datasource/puppet-forge/index.ts

```diff
@@ -0,0 +1,55 @@
+import { logger } from '../../../logger';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+import { PUPPET_FORGE } from './common';
+import type { PuppetModule } from './types';
+
+export class PuppetForgeDatasource extends Datasource {
+  static id = 'puppet-forge';
+
+  constructor() {
+    super(PuppetForgeDatasource.id);
+  }
+
+  override readonly defaultRegistryUrls = [PUPPET_FORGE];
+
+  async getReleases({
+    packageName,
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    // https://forgeapi.puppet.com
+    const moduleSlug = packageName.replace('/', '-');
+    const url = `${registryUrl}/v3/modules/${moduleSlug}`;
+
+    let moduleResponse;
+    try {
+      moduleResponse = await this.http.get(url);
+    } catch (err) {
+      logger.warn(
+        { err },
+        `ignore dependency ${packageName} because of faulty response for ${url}`
+      );
+      return null;
+    }
+
+    const module: PuppetModule = JSON.parse(moduleResponse.body);
```

Comment:
```suggestion
    const { body: module } = await this.http.getJson<PuppetModule>(url);
```

use direct json http function

Reply:
I like to remove code.. :)